Add extra challenges

dancer-yuma opened the pull request:
# What I did

- add challenges for each unit
- add challenge tests for them
- add a solution repo `python-01-solutions` and solutions for them (https://github.com/ms1-learner/python-01-solutions/pull/1)

# What I want reviewers to check

- if the level of difficulty is appropriate
- grammar
- if tests work as expected
- if I have not added solutions accidentally

# Notes

- we need to reconsider which one should be for live challenge sessions
- these need to be localized
- MS1 challenge guides also need updates

xvarc-wp commented on 02_creating_python_files_and_control_flow/03_challenge/count_vowels.py:
@dancer-yuma I think we should put "#your code here" in empty python files.

dancer-yuma replied:
I will work on it!

xvarc-wp reviewed (approved):
Great work yuma!

dancer-yuma commented on 01_introduction_to_software_development_in_python/02_challenge/test_main.py:
@xvarc-wp They are float when the expression uses more than one division operator or the operands are float. But `5 == 5.0` is `True` anyways. I don't think this will be an issue!

```
>>> 2 + 3
5
>>> 2 + 3.0
5.0
>>> 10 / 2
5.0
>>> 2 + 3 == 10 / 2
True
```

dancer-yuma commented:
@xvarc-wp 
Could you review again?
I removed `?` and added `# Your code here` to all the files in https://github.com/ms1-learner/python-01-en/compare/d9d9da0...16711f3

xvarc-wp reviewed (approved):
LGTM!
